cc/mcts_node.cc

```diff
@@ -129,24 +129,51 @@ Coord MctsNode::GetMostVisitedMove() const {
   return c;
 }
 
+void MctsNode::ReshapeFinalVisits() {
+  Coord best = GetMostVisitedMove();
+  float U_common = U_scale() * std::sqrt(1.0f + N());
+  float to_play = position.to_play() == Color::kBlack ? 1 : -1;
+  float best_cas =
+      CalculateSingleMoveChildActionScore(to_play, U_common, uint16_t(best));
+
+  //int total = 0;
+  for (int i = 0; i < kNumMoves; ++i) {
+    if (i == uint16_t(best)) {
+      continue;
+    }
+
+    // Change N_child to the smallest value that satisfies the inequality
```

Comment:
add a note about why we do this.

"When we explored this position we had uncertainty about the value of this move,
now after we have expanded this node, we change the visit count to reflect how many
visits we would have given it with our updated understand of it's value (and the value of the best move)"